tests/test_prolongation.cpp

```diff
@@ -0,0 +1,297 @@
+                           std::to_string(i) + ")";
+                ;
+            }
+            else if (i % 2 == 0 && j % 2 != 0) {
+            // as numbering in radius (r_j) is odd, we have a fine node with
+            // coarse neighbors at (r_j - h_{j-1}, theta_i) and (r_j+h_j, theta_i )
+                double h_pm1 = p_level.r[j] - p_level.r[j - 1]; //h_{p-1}
+                double h_p   = p_level.r[j + 1] - p_level.r[j]; //h_p
+
+                double v1 = u_test[(j - 1) / 2 * ctheta_int + (i / 2)]; // left coarse node in r
+                double v2 = u_test[(j + 1) / 2 * ctheta_int + (i / 2)]; // right coarse node in r
+
+                double val = (h_p * v1 + h_pm1 * v2);
+
+                EXPECT_NEAR(val, (h_p + h_pm1) * sol[j * p_level.ntheta_int + i], 1e-8) //compare values
+                    << "Bilinear Prolongation fails for Index (r,theta) : (" + std::to_string(j) + "," +
+                           std::to_string(i) + ")";
+                ;
+            }
+            else // both are fine nodes
+            {
+                double h_pm1 = p_level.r[j] - p_level.r[j - 1];
+                double h_p   = p_level.r[j + 1] - p_level.r[j];
+
+                double k_qm1 = p_level.theta[i] - p_level.theta[i - 1];
+                double k_q =
+                    (i < p_level.ntheta_int - 1) ? p_level.theta[i + 1] - p_level.theta[i] : 2 * PI - p_level.theta[i];
+
+                /*the stencil-corners corresponding to the values (r_j,theta_i)*/
+                /*bottom corresponds to lower indices in theta direction. left to lower indices in radius direction*/
+
+                double bottom_left;
+                double top_left;
+                double bottom_right;
+                double top_right;
+                ASSERT_NE(p_level.nr_int - 1, 1);
+
+                if (i < p_level.ntheta - 1) {
+                    bottom_left  = u_test[((j - 1) / 2) * ctheta_int + (i - 1) / 2];
+                    top_left     = u_test[((j - 1) / 2) * ctheta_int + (i + 1) / 2];
+                    bottom_right = u_test[((j + 1) / 2) * ctheta_int + (i - 1) / 2];
+                    top_right    = u_test[((j + 1) / 2) * ctheta_int + (i + 1) / 2];
+                }
+                else {
+                    bottom_left  = u_test[((j - 1) / 2) * ctheta_int + (i - 1) / 2];
+                    top_left     = u_test[((j - 1) / 2) * ctheta_int];
+                    bottom_right = u_test[((j + 1) / 2) * ctheta_int + (i - 1) / 2];
+                    top_right    = u_test[((j + 1) / 2) * ctheta_int];
+                }
+
+                double val = ((h_p * k_q * bottom_left) + (h_p * k_qm1 * top_left) + (h_pm1 * k_q * bottom_right) +
+                              (h_pm1 * k_qm1 * top_right)); //calculate value as in the paper
+
+                EXPECT_NEAR(val, (h_p + h_pm1) * (k_q + k_qm1) * sol[j * p_level.ntheta_int + i], 1e-8)
```

Comment:
```suggestion
                EXPECT_NEAR(val, (h_p + h_pm1) * (k_q + k_qm1) * sol[j * p_level.ntheta_int + i], 1e-12)
```
Can we make this succeed with 1e-12 too?